src/main.cpp

```diff
@@ -8303,7 +8303,7 @@ bool MemorizeMessage(const CTransaction &tx, double dAmount, std::string sRecipi
                                 if (!(sMessageType=="project" || sMessageType=="projectmapping" || sMessageType=="beacon" ))
                                 {
                                     WriteCache(sMessageType,sMessageKey+";Recipient",sRecipient,nTime);
-                                    WriteCache(sMessageType,sMessageKey+";BurnAmount",RoundToString(dAmount,2),nTime);
+                                    WriteCache(sMessageType,sMessageKey+";BurnAmount",ToString(dAmount),nTime);
```

Comment:
But I do not like this increase of decimal places.
The appcache is suboptimal solution to runtime data storage, but it is how it is until someone changes it.
Increase of decimal places from 2 to 8 would mean increase of anon memory usage. The wallet is already using more than enough anon memory.

Reply:
I'll be replacing `stringstream` with `sprintf` in `ToString` which is both faster and discards trailing zeroes (IIRC). The cache does not use a whole lot of memory, but it will be an improvement.